pkg/security/api/types.go

```diff
@@ -0,0 +1,86 @@
+package api
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+// PodSecurityPolicySubjectReview checks whether a particular user/SA tuple can create the PodSpec.
+type PodSecurityPolicySubjectReview struct {
```

Comment:
> I personally can check a template / pod spec is allowed when run under a DC / RC (i.e. For a service account)

`PodSecurityPolicyReview` checks if SA (or which SAs) can create the pod





> I personally can check a pod spec I'm about to create

`PodSecurityPolicySelfSubjectReview` different name for different acl.  any editor can check for themselves, but only a project admin can check for a particular user (its not the editor's business).



> the deployment controller can check the pod spec and not deploy / hold / fail a deployment until the pod spec would work.

`PodSecurityPolicyReview` same as the first.  the controller just has rights to use it in any namespace.